scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/handler/MainLambdaHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.schedulingmanager.deployment.handler;
+
+import com.amazonaws.blox.schedulingmanager.deployment.DeploymentWorkflowApplication;
+import com.amazonaws.blox.schedulingmanager.deployment.exception.HandlerNotFoundException;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MainLambdaHandler extends DeploymentWorkflowApplication
+    implements RequestStreamHandler {
+
+  @Override
+  public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context)
+      throws IOException {
+
+    final String inputJson;
+
+    try (BufferedReader buffer = new BufferedReader(new InputStreamReader(inputStream))) {
+      inputJson = buffer.lines().collect(Collectors.joining("\n"));
+    }
+
+    // find handler matching the lambda function alias
+    final Map<String, Handler> handlers = applicationContext.getBean("handlers", Map.class);
```

Comment:
Can we inject these into the handler with `@Autowired` instead of reaching into the context for it manually? This line generates an `unchecked assignment` warning as is, and it looks like the only way to avoid that is to move it to an injected field/constructor parameter.

Reply:
This is the main function that initializes the beans, I don't think autowired can find beans before they're initialized, unless you know how to do that?

Reply:
See comment above about potentially not injecting those. Otherwise, 3 tiers of handlers could take care of it:

- `MainHandler`: instantiates spring context, and instantiates `DispatcherHandler`
- `DispatcherHandler`: uses the injected `Map` of `StepHandler`s to look up the appropriate handler for the given alias
- `*StepHandler`s: actually handles the step(s)

Reply:
Yep that's what I was suggesting above by having a separate class find the correct handler.

Reply:
👍